sdk/python/kfp/compiler/compiler_utils_test.py

```diff
@@ -53,6 +53,82 @@ def test_additional_input_name_for_pipeline_channel(self, channel,
             expected,
             compiler_utils.additional_input_name_for_pipeline_channel(channel))
 
+    @parameterized.parameters(
+        {
+            'data': [{
+                'container_spec': {
+                    'image_uri':
+                        'gcr.io/ml-pipeline/google-cloud-pipeline-components:2.5.0',
+                    'command': ['echo'],
+                    'args': ['foo']
+                },
+                'machine_spec': {
+                    'machine_type':
+                        pipeline_channel.PipelineParameterChannel(
+                            name='Output',
+                            channel_type='String',
+                            task_name='machine-type'),
+                    'accelerator_type':
+                        pipeline_channel.PipelineParameterChannel(
+                            name='Output',
+                            channel_type='String',
+                            task_name='accelerator-type'),
+                    'accelerator_count':
+                        1
+                },
+                'replica_count': 1
+            }],
+            'old_value':
+                '{{channel:task=machine-type;name=Output;type=String;}}',
+            'new_value':
+                '{{$.inputs.parameters['
+                'pipelinechannel--machine-type-Output'
+                ']}}',
+            'expected': [{
+                'container_spec': {
+                    'image_uri':
+                        'gcr.io/ml-pipeline/google-cloud-pipeline-components:2.5.0',
+                    'command': ['echo'],
+                    'args': ['foo']
+                },
+                'machine_spec': {
+                    'machine_type':
+                        '{{$.inputs.parameters['
+                        'pipelinechannel--machine-type-Output'
+                        ']}}',
+                    'accelerator_type':
+                        pipeline_channel.PipelineParameterChannel(
+                            name='Output',
+                            channel_type='String',
+                            task_name='accelerator-type'),
+                    'accelerator_count':
+                        1
+                },
+                'replica_count': 1
+            }],
+        },
+        {
+            'data': [{
+                0: 'a',
+                1: 'b',
+                2: 'c',
+                3: ['d']
+            }],
+            'old_value': 'd',
+            'new_value': 'dd',
```

Comment:
The old and new values doesn't seem to be testing the real case scenario?

Reply:
That testcase was for some simply dummy values just to verify behavior of `recursive_replace()`, I can remove if you think its redundant?

Reply:
Removed testcase.